### PR TITLE
[EAM-168] Add inventory web services

### PR DESCRIPTION
**Pull request by pedroamarocern**
Adds the following inventory web services (corresponding to the "Physical Inventory" functionality in Infor EAM):
- Create Physical Inventory
- Read Physical Inventory
- Update Physical Inventory
- Read Physical Inventory Line
- Update Physical Inventory Line
- Read Default Physical Inventory

### CODE REVIEW

File: src/main/java/ch/cern/eam/wshub/core/services/material/PhysicalInventoryService.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Operation(logOperation = INFOR_OPERATION.INVENTORY_R)
+    PhysicalInventory readPhysicalInventory(InforContext context, String code)
+        throws InforException;
+
+    @Operation(logOperation = INFOR_OPERATION.INVENTORY_U)
+    PhysicalInventory updatePhysicalInventory(InforContext context, PhysicalInventory physicalInventory)
+        throws InforException;
+
+    @Operation(logOperation = INFOR_OPERATION.INVENTORY_R)
```
Comment (ricardo-j-almeida) [inline src/main/java/ch/cern/eam/wshub/core/services/material/PhysicalInventoryService.java:23]: Each different operation should have a different key for the logger

File: src/main/java/ch/cern/eam/wshub/core/services/material/impl/PhysicalInventoryServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    @Override
+    public PhysicalInventory readPhysicalInventory(InforContext context, String code) throws InforException {
+        ResultData resultData = getInventoryResultData(context, code);
+        PhysicalInventory physicalInventory = tools.getInforFieldTools().transformInforObject(
+            new PhysicalInventory(),
+            resultData.getInventoryTransaction());
+
+        if(resultData.getCREATEDBY() != null)
+            physicalInventory.setCreatedBy(resultData.getCREATEDBY().getUSERCODE());
```
Comment (ricardo-j-almeida) [inline src/main/java/ch/cern/eam/wshub/core/services/material/impl/PhysicalInventoryServiceImpl.java:69]: if statements that do not fully appear on one line must have curly braces

File: src/main/java/ch/cern/eam/wshub/core/tools/DataTypeTools.java [hunk 3]
```diff
@@ -386,4 +386,8 @@ public static String toCodeString(CLASSID_Type classIdType) {
     public static String toCodeString(LOCATIONID_Type locationIdType) {
         return locationIdType == null ? null : locationIdType.getLOCATIONCODE();
     }
+
+    public static Long convertBigIntegerToLong(BigInteger value) {
+        return value.longValueExact();
```
Comment (ricardo-j-almeida) [inline src/main/java/ch/cern/eam/wshub/core/tools/DataTypeTools.java:391]: NPE

File: src/main/java/ch/cern/eam/wshub/core/services/material/PhysicalInventoryService.java (hunk 1 above)
Comment (pedroamarocern) [reply src/main/java/ch/cern/eam/wshub/core/services/material/PhysicalInventoryService.java]: https://github.com/cern-eam/eam-wshub-core/pull/22/files#diff-6146b6776d5383f5c8a97c71c1f08169R11

File: src/main/java/ch/cern/eam/wshub/core/tools/DataTypeTools.java (hunk 3 above)
Comment (pedroamarocern) [reply src/main/java/ch/cern/eam/wshub/core/tools/DataTypeTools.java]: https://github.com/cern-eam/eam-wshub-core/pull/22/files#diff-cf703fd85afb83ccc960da1b1714dbf7R390

**Pull request merged**